pkg/identity/service.go

```diff
@@ -56,9 +67,92 @@ type (
 	NotAuthenticated struct{}
 )
 
+// Initialize loads the issuer certs from disk so it can start service CSRs to proxies
+func (svc *Service) Initialize() error {
+	credentials, err := svc.loadCredentials()
+	if err != nil {
+		return err
+	}
+	svc.updateIssuer(credentials)
+	return nil
+}
+
+func (svc *Service) updateIssuer(newIssuer tls.Issuer) {
+	svc.issuerMutex.Lock()
+	svc.issuer = &newIssuer
+	log.Debug("Issuer has been updated")
+	svc.issuerMutex.Unlock()
+}
+
+// Run reads from the issuer and error channels and reloads the issuer certs when necessary
+func (svc *Service) Run(issuerEvent <-chan struct{}, issuerError <-chan error) {
+	for {
+		select {
+		case <-issuerEvent:
+			if err := svc.Initialize(); err != nil {
+				message := fmt.Sprintf("skipping issuer update as certs could not be read from disk: %s", err)
+				log.Warn(message)
+				svc.recordEvent(v1.EventTypeWarning, eventTypeSkipped, message)
+			} else {
+				message := "updated identity issuer"
+				log.Infof(message)
+				svc.recordEvent(v1.EventTypeNormal, eventTypeUpdated, message)
```

Comment:
Nit: Can we use `strings.ToUpper(message)` to make the output consistent with the rest of the k8s events? This is what it currently looks like on my cluster, where the `IssuerUpdated` looks a bit different from the rest:
```
⚡   k -n linkerd get ev | grep -i issuer                                                                        
10m         Normal    CertificateIssued   certificaterequest/linkerd-identity-issuer-860701636   Certificate fetched from issuer successfully
5m41s       Normal    CertificateIssued   certificaterequest/linkerd-identity-issuer-860701636   Certificate fetched from issuer successfully
41s         Normal    CertificateIssued   certificaterequest/linkerd-identity-issuer-860701636   Certificate fetched from issuer successfully
10m         Normal    GeneratedKey        certificate/linkerd-identity-issuer                    Generated a new private key
41s         Normal    Requested           certificate/linkerd-identity-issuer                    Created new CertificateRequest resource "linkerd-identity-issuer-860701636"
41s         Normal    Issued              certificate/linkerd-identity-issuer                    Certificate issued successfully
37s         Normal    IssuerUpdated       deployment/linkerd-identity                            updated identity issuer
```
Also, on line 95 above.